app/commands/v2/discard_draft.rb

```diff
@@ -13,8 +13,18 @@ def call
         increment_live_lock_version if live
 
         after_transaction_commit do
-          delete_draft_from_draft_content_store(draft_path)
-          send_live_to_draft_content_store(live)
+          if live
+            live_path = Location.find_by!(content_item: live).base_path
+
+            if live_path == draft_path
```

Comment:
I tried to optimise for making it clear what actions are taken in each case. I tried a few different variations, but the above it the best I could come up with.

In particular, the suggestion you give above looks like it always sends the "live" version to the draft content store, which is not always the case.